src/Sentry/Platforms/Android/LogCatAttachmentEventProcessor.cs

```diff
@@ -0,0 +1,96 @@
+using Sentry.Extensibility;
+using Runtime = Java.Lang.Runtime;
+using Android.Content;
+using Application = Android.App.Application;
+
+namespace Sentry.Android;
+
+internal class LogCatAttachmentEventProcessor : ISentryEventProcessorWithHint
+{
+    private static bool SendLogcatLogs = true;
+    private readonly LogCatIntegrationType _logCatIntegrationType;
+    private readonly IDiagnosticLogger? _diagnosticLogger;
+
+    public LogCatAttachmentEventProcessor(LogCatIntegrationType logCatIntegrationType, IDiagnosticLogger? diagnosticLogger)
+    {
+        _logCatIntegrationType = logCatIntegrationType;
+        _diagnosticLogger = diagnosticLogger;
+    }
+
+    public SentryEvent Process(SentryEvent @event, Hint hint)
+    {
+        // If sending has failed once, we have to disable this feature to prevent infinite loops and to allow the SDK to work otherwise
+        if (!SendLogcatLogs)
+        {
+            return @event;
+        }
+
+        // The logcat command only works on Android API 23 and above
+        if (!OperatingSystem.IsAndroidVersionAtLeast(23))
+        {
+            SendLogcatLogs = false;
+            return @event;
+        }
+
+        try
+        {
+            if (_logCatIntegrationType != LogCatIntegrationType.All && (@event.SentryExceptions?.Any() ?? false))
```

Comment:
https://github.com/getsentry/sentry-dotnet/blob/27ede88985fcc92ce2f7f8ae53ba11cfd805b2f5/src/Sentry/SentryEvent.cs#L182C19-L182C31

```suggestion
            if (_logCatIntegrationType != LogCatIntegrationType.All && !@event.HasException())
```